addons/godot-twicil/godot-twicil/helpers/twitch_message_wrapper.gd

```diff
@@ -0,0 +1,87 @@
+class_name TwitchMessage
+
+enum EmoteType {TWITCH, BTTV, FFZ}
+
+var chat_message: IrcChatMessage
+#[
+#    {
+#        'code': 'emote_code',
+#        'id': 'emote_id'
+#        'type': 0    # EmoteType enum
+#    },
+#    ...
+#]
+#
+var emotes: Array
+
+
+func _init(server_irc_message: TwitchIrcServerMessage):
+    chat_message = MessageWrapper.wrap(server_irc_message)
+
+    emotes.clear()
+
+    __parse_twitch_emotes(server_irc_message.message_prefix)
+
+
+# private
+func __parse_twitch_emotes(message_prefix: String):
+    var prefix_params := message_prefix.split(';', false)
+    var emotes_param: String
+
+    for param in prefix_params:
+        if (param as String).begins_with('emotes'):
+            var emotes_prefix_param: Array = (param as String).split('=', false, 1)
+
+            if emotes_prefix_param.size() <= 1:
+                return
+
+            emotes_param = emotes_prefix_param[1]
+
+            for emote in emotes_param.split('/', false):
+                var emote_data: Array = emote.split(':', false)
+                var id := int(emote_data[0])
+
+                var positions: Array = emote_data[1].split(',', false)[0].split('-', false)
+
+                var start := int(positions[0])
+                var end := int(positions[1])
+
+                var code: String = chat_message.text.substr(start, end - start + 1)
+
+                emotes.append({
+                    'id': id,
+                    'code': code,
+                    'type': EmoteType.TWITCH
+                })
+
+func parse_bttv_emotes(available_emotes: Dictionary) -> void:
+    var message: String = ' ' + chat_message.text + ' '
+
+    for emote_code in available_emotes:
+        var parse_emote_code: String = ' ' + emote_code + ' '
+
+        if message.find(parse_emote_code) >= 0:
+            emotes.append({
+                'id': available_emotes.get(emote_code),
```

Comment:
Maybe represent `available_emotes` as an own class to unify the retrieval of data, in order to join `parse_bttv_emotes` and `parse_ffz_emotes` functions.